Formula/e/egctl.rb

```diff
@@ -0,0 +1,30 @@
+class Egctl < Formula
+  desc "Command-line utility for operating Envoy Gateway"
+  homepage "https://gateway.envoyproxy.io/"
+  url "https://github.com/envoyproxy/gateway/archive/refs/tags/v1.0.2.tar.gz"
+  sha256 "05406182dc165513925cf60722582613d4de9ea789d60e014e6da456bb229f65"
+  license "Apache-2.0"
+  head "https://github.com/envoyproxy/gateway.git", branch: "main"
+
+  livecheck do
+    url :stable
+    strategy :github_latest
+  end
+
+  depends_on "go" => :build
+
+  def install
+    ldflags = %W[
+      -s -w
+      -X github.com/envoyproxy/gateway/internal/cmd/version.envoyGatewayVersion=#{version}
+      -X github.com/envoyproxy/gateway/internal/cmd/version.gitCommitID=#{tap.user}
+    ]
+    system "go", "build", *std_go_args(ldflags:), "./cmd/egctl"
+
+    generate_completions_from_executable(bin/"egctl", "completion")
+  end
+
+  test do
+    assert_equal version.to_s, shell_output("#{bin}/egctl version --remote=false").strip
```

Comment:
Please use the preferred commit-message style for homebrew/core. We put the name of the formula first in commit-message headings.

For new formulae:

> At Homebrew, we like to put the name of the formula up front like so: `foobar 7.3 (new formula)`.

For existing formulae:

> The preferred commit message format for simple version updates is `foobar 7.3` and for fixes is `foobar: fix flibble matrix.`.

Refer to the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit) for more details. Also, when making further changes to your pull request, adhere to the following guidelines to make sure that @paperchalice can merge your commits:

> - One formula per commit; one commit per formula.
> - Keep merge commits out of the pull request.